tools/dumper/source/main.c

```diff
@@ -64,8 +64,11 @@ void kpatch_enablemapself(struct thread *td)
     uint8_t* kernel_base = &((uint8_t*)read_msr(0xC0000082))[-0x1C0];
     uint8_t* map_self_patch1 = &kernel_base[0x117B0];
     uint8_t* map_self_patch2 = &kernel_base[0x117C0];
-    /* map_self_patch3 = &kernel_base[0x13F03F]; (5.05) */
+#ifdef VERSION_505
+    uint8_t* map_self_patch3 = &kernel_base[0x13F03F]; //5.05
+#else
```

Comment:
The line below only works in 5.00. To prevent accidentally patching the wrong offset, let's change the current `#else` below by `#elif VERSION_500`.

Still, I'd add an *else* case to trigger a compiler error:
```c
#else
    #error "Target firmware not yet supported."
#endif
```

Reply:
I addressed this comment. The problem is that this patch works differently for different firmware versions. For 4.55 there is a patch function here: https://github.com/AlexAltea/orbital/blob/8137c1c90082b9a06d4115b2bfb9081c87500bc1/tools/dumper/source/ksdk_455.inc#L26
And I am not sure if the offsets and patches are correct for 5.05 either (at least for me the console crashes, but that also might be a different cause). 
To do the patching the right way we should add defines for all firmwares and use them accordingly.
I don`t really know where all these offsets came from. Should I add a TODO here or something?

Reply:
I see. Well, it can be left for a future patch then. :-)